app/ui/src/app/dashboard/dashboard_integrations/dashboard-integrations.component.ts

```diff
@@ -1,53 +1,73 @@
+import { DonutComponent, DonutConfig } from 'patternfly-ng';
 
-const category = getCategory('Dashboard');
+import { log } from '@syndesis/ui/logging';
+import {
+  Connection,
+  Connections,
+  IntegrationOverview,
+  IntegrationOverviews,
+  IntegrationSupportService
+} from '@syndesis/ui/platform';
 
 @Component({
   selector: 'syndesis-dashboard-integrations',
   templateUrl: './dashboard-integrations.component.html',
   styleUrls: ['./dashboard-integrations.component.scss']
 })
-export class DashboardIntegrationsComponent implements OnChanges {
-  chartData: number[];
-  @Input() integrations: IntegrationOverviews;
+export class DashboardIntegrationsComponent implements OnInit {
+
+  integrationsObserver: Observable<IntegrationOverviews>;
+  integrations: Array<IntegrationOverview>;
+  loading = true;
+
   @Input() connections: Connections;
-  @Input() integrationsLoading: boolean;
   @Input() connectionsLoading: boolean;
 
-  doughnutChartLabels = ['Published', 'Draft', 'Unpublished'];
-
-  get doughnutChartData() {
-    return this.chartData;
-  }
+  integrationChartData: any[] = [
+    ['Published', 0],
+    ['Draft', 0],
+    ['Unpublished', 0]
+  ];
 
-  doughnutChartType = 'doughnut';
-  doughnutChartLegend = false;
-  doughnutChartColors = [
-    {
-      backgroundColor: [
-        '#0088CE', // PatternFly Blue 400, Published
-        '#EC7A08', // PatternFly Orange 400, Draft
-        '#D1D1D1' // PatternFly Black 300, Unpublished
-      ]
+  integrationsChartConfig: DonutConfig = {
+    chartHeight: 120,
+    chartId: 'integrationsCounter',
+    colors: {
+      Published: '#0088CE',   // PatternFly Blue 400, Published
+      Draft: '#EC7A08',       // PatternFly Orange 400, Draft
+      Unpublished: '#D1D1D1'  // PatternFly Black 300, Unpublished
+    },
+    donut: {
+      title: 'Integrations'
+    },
+    legend: {
+      show: true,
+      position: 'right'
     }
-  ];
-  doughnutChartOptions: any = {
-    cutoutPercentage: 75
   };
 
-  constructor(public route: ActivatedRoute, private router: Router) {
-    this.chartData = [0, 0, 0];
+  constructor(
+    public route: ActivatedRoute,
+    private router: Router,
+    private integrationSupportService: IntegrationSupportService
+  ) {
   }
 
-  ngOnChanges(changes: any) {
-    this.chartData[0] = this.countActiveIntegrations();
-    this.chartData[1] = this.countDraftIntegrations();
-    this.chartData[2] = this.countInactiveIntegrations();
+  ngOnInit() {
+    this.integrationsObserver = this.integrationSupportService.watchOverviews();
+    this.integrationsObserver.subscribe(integrations => {
+      this.integrations = integrations;
+      this.loading = false;
+      this.integrationChartData = [
+        [`Published`, this.countActiveIntegrations()],
+        [`Draft`, this.countDraftIntegrations()],
+        [`Unpublished`, this.countInactiveIntegrations()]
+      ];
+    });
```

Comment:
`integrationSupportService.watchOverviews()` is fed by the Socket Messaging layer. That means it will keep "emitting" values that will be funneled down through the subscription defined here even after the component is destroyed and disposed. Therefore this subscription should be terminated once `DashboardIntegrationsComponent` is destroyed.

Add a private subscription class member like this to the top, right after the last class public members (`integrationsChartConfig`):

```typescript
private integrationOverviewsSubscription: Subscription;
````

Assign the above field to the subscription:

```typescript
this.integrationOverviewsSubscription = this.integrationsObserver.subscribe(integrations => {
...
```

and then implement the `OnDestroy` interface in the class and append this lifecycle handler right after `ngOnInit()`:

```typescript
ngOnDestroy() {
  if (this.integrationOverviewsSubscription) {
    this.integrationOverviewsSubscription.unsubscribe();
  }
}
```

Reply:
Ooh, missed that, good catch!  I withdraw my approval :-)